### ISSUE TITLE
no data found

### ISSUE THREAD

**Issue by HB9VQQ** [coulisse/spiderweb#36]
Until this morning all was working fine until I noticed there are no new Spots incoming. In the log I found.

2023-02-23 09:46:23,076 [INFO] (_internal.py) 84.73.221.190 - - [23/Feb/2023 09:46:23] "GET /static/images/icons/spider_ico_master.svg HTTP/1.1" 200 -
2023-02-23 09:46:23,764 [INFO] (_internal.py) 84.73.221.190 - - [23/Feb/2023 09:46:23] "GET /static/images/icons/favicon.ico HTTP/1.1" 200 -
2023-02-23 09:46:32,010 [WARNING] (webapp.py) **no data found**

**Startup looking good**

2023-02-23 09:50:13,290 [INFO] (webapp.py) Start
2023-02-23 09:50:13,292 [INFO] (cty.py) CTY: start initialization
2023-02-23 09:50:13,292 [INFO] (cty.py) /spiderweb/lib/../static/data/cty_wt_mod.dat updated (6.0 days), is not necessary to download it
2023-02-23 09:50:13,296 [INFO] (cty.py) number of lines reads: 8368
2023-02-23 09:50:13,296 [INFO] (cty.py) number of valid lines: 7398
2023-02-23 09:50:13,436 [INFO] (cty.py) number of entities: 346
2023-02-23 09:50:13,436 [INFO] (cty.py) number of single alias: 27834
2023-02-23 09:50:14,311 [INFO] (cty.py) memory used for prefix: 1310792 bytes
2023-02-23 09:50:14,311 [INFO] (cty.py) CTY: initialization complete
2023-02-23 09:50:14,312 [INFO] (qry.py) [Errno 2] No such file or directory: '../cfg/config.json'
2023-02-23 09:50:14,312 [INFO] (qry.py) trying with other path...
2023-02-23 09:50:14,312 [INFO] (qry.py) config file loaded
2023-02-23 09:50:14,315 [INFO] (qry.py) db connection pool created
2023-02-23 09:50:14,315 [INFO] (adxo.py) connection to: http://dxcal.kj4z.com/dxcal
2023-02-23 09:50:14,374 [INFO] (adxo.py) number ADXO events: 62
2023-02-23 09:50:14,375 [INFO] (plot_data_provider.py) Class: ContinentsBandsProvider init start
2023-02-23 09:50:14,375 [INFO] (plot_data_provider.py) Class: ContinentsBandsProvider refresh data
2023-02-23 09:50:14,375 [INFO] (plot_data_provider.py) Start
2023-02-23 09:50:14,376 [INFO] (plot_data_provider.py) doing query...
2023-02-23 09:50:14,378 [WARNING] (plot_data_provider.py) no data found
2023-02-23 09:50:14,378 [INFO] (plot_data_provider.py) query done
2023-02-23 09:50:14,379 [INFO] (plot_data_provider.py) Class: ContinentsBandsProvider init end
2023-02-23 09:50:14,379 [INFO] (plot_data_provider.py) Class: SpotsPerMounthProvider init start
2023-02-23 09:50:14,379 [INFO] (plot_data_provider.py) Class: SpotsPerMounthProvider refresh data
2023-02-23 09:50:14,379 [INFO] (plot_data_provider.py) Start
2023-02-23 09:50:14,379 [INFO] (plot_data_provider.py) doing query...
2023-02-23 09:50:14,465 [INFO] (plot_data_provider.py) query done
2023-02-23 09:50:14,466 [INFO] (plot_data_provider.py) Class: SpotsPerMounthProvider init end
2023-02-23 09:50:14,466 [INFO] (plot_data_provider.py) Class: SpotsTrend init start
2023-02-23 09:50:14,466 [INFO] (plot_data_provider.py) Class: SpotsTrend refresh data
2023-02-23 09:50:14,466 [INFO] (plot_data_provider.py) Start
2023-02-23 09:50:14,466 [INFO] (plot_data_provider.py) doing query...
/spiderweb/lib/qry.py:77: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  self.__data = pd.read_sql(qs, con=cnx)
2023-02-23 09:50:14,529 [INFO] (plot_data_provider.py) query done
2023-02-23 09:50:14,537 [INFO] (plot_data_provider.py) Class: SpotsTrend init end
2023-02-23 09:50:14,537 [INFO] (plot_data_provider.py) Class: HourBand init start
2023-02-23 09:50:14,538 [INFO] (plot_data_provider.py) Class: HourBand refresh data
2023-02-23 09:50:14,538 [INFO] (plot_data_provider.py) Start
2023-02-23 09:50:14,538 [INFO] (plot_data_provider.py) doing query...
2023-02-23 09:50:14,644 [INFO] (plot_data_provider.py) query done
2023-02-23 09:50:14,644 [INFO] (plot_data_provider.py) Class: HourBand init end
2023-02-23 09:50:14,644 [INFO] (plot_data_provider.py) Class: WorldDxSpotsLive init start
2023-02-23 09:50:14,644 [INFO] (plot_data_provider.py) Class: WorldDxSpotsLive refresh data
2023-02-23 09:50:14,645 [INFO] (plot_data_provider.py) Start
2023-02-23 09:50:14,645 [INFO] (plot_data_provider.py) doing query...
2023-02-23 09:50:14,651 [WARNING] (plot_data_provider.py) no data found
2023-02-23 09:50:14,651 [INFO] (plot_data_provider.py) query done
2023-02-23 09:50:14,657 [INFO] (plot_data_provider.py) Class: WorldDxSpotsLive init end
Serving Flask app 'webapp'

Any idea?

**Comment by HB9VQQ** [comment]
Apparently it is mariadb related

# journalctl -u mariadb -f
 mariadbd[109316]: 2023-02-23 10:06:21 22 [Warning] Aborted connection 22 to db: 'dxcluster' user: 'root' host: 'localhost' (Got an error reading communication packets)

**Comment by coulisse** [comment]
Hello 
yes it is apparently related to mariadb.

could you check if mariadb is running (systemctl status mariadb)

Moreover I see in your message that you are connecting as root user to dxcluster.

I think is better to define a specific user and assign it to dxcluster:

for example
grant all privileges on dxcluster.* to webdb@localhost identified by 'pswd';

By the way you have to modify config.json with new user and password.

Waiting for your feedback
Best regards
Corrado

**Comment by HB9VQQ** [comment]
Thanks, 

yes mariadb is running
![image](https://user-images.githubusercontent.com/75934980/220894222-25c94008-f1a2-4e06-b8e5-676186e5c80f.png)

It was running without problems using root. I will change it as soon as it is working again.

I deleted the database 'dxcluster' and was hoping it is re-created by restarting dxspider, That's not the case unfortunately. Then I tried to follow https://github.com/coulisse/spiderweb/blob/master/docs/DB_ISSUES.md but the command to create the table/coloums returned errors.
